x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/common/BlobCacheBufferedIndexInput.java

```diff
@@ -352,6 +353,23 @@ protected final IndexInput trySliceBuffer(String name, long sliceOffset, long sl
         return null;
     }
 
+    @Nullable
+    protected final IndexInput tryCloneBuffer() {
+        if (buffer.limit() == length && bufferStart == 0) {
```

Comment:
Isn't this check already covered by the range check in trySliceBuffer?

Reply:
Sort of, the `0` start offset isn't and we need it. I figured it's a little easier to read this by adding the other check and it saves cycles for range object allocations (maybe). But could probably do without the first condition indeed, I'll leave it for now hope that's ok.